extensions/amp-apester-media/0.1/amp-apester-media.js

```diff
@@ -0,0 +1,141 @@
+ * limitations under the License.
+ */
+
+import {CSS} from '../../../build/amp-apester-media-0.1.css';
+import {user} from '../../../src/log';
+import {loadPromise} from '../../../src/event-helper';
+import {setStyles} from '../../../src/style';
+import {listenFor} from '../../../src/iframe-helper';
+
+/**
+ * AMP Apester-media
+ */
+class AmpApesterMedia extends AMP.BaseElement {
+
+  // /** @override */
+  // isLayoutSupported() {
+  //   return true;// (layout == Layout.CONTAINER);
+  // }
+
+  /** @override */
+  viewportCallback() {
+    // TODO: if amp left the viewport, send iframe post message that it has been seen.
+  }
+
+  /** @override */
+  buildCallback() {
+    /**
+     * @private {?String}
+     */
+    this.displayBaseUrl_ = window.location.protocol +
+      '//stage3-renderer.qmerce.com'; //renderer.qmerce.local.com
+
+    /**
+     * @private {?Element}
+     */
+    this.mediaAttribute_ = user.assert(
+      (this.element.getAttribute('data-apester-media-id') ||
+      this.element.getAttribute('data-apester-channel-token')),
+      'Either the data-apester-media-id or the data-apester-channel-token ' +
+      'attributes must be specified for <amp-apester-media> %s',
+      this.element);
+
+    /**
+     * @private {?Element}
+     */
+    this.iframe_ = null;
+    /**
+     * @private {?Promise}
+     */
+    this.iframePromise_ = null;
+
+    //TODO change to setStyle?
+    this.element.classList.add('-amp-apester-container');
+  }
+
+  /** @override */
+  firstLayoutCompleted() {
+    // Do not hide placeholder
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.setAttribute('scrolling', 'no');
+    iframe.src = this.displayBaseUrl_ + '/interaction/'
+      + this.mediaAttribute_;
+    iframe.height = this.element.getAttribute('height');
+    iframe.classList.add('-amp-apester-iframe');
+    this.applyFillContent(iframe);
+
+    setStyles(iframe, {
+      'opacity': 0,
+    });
+
+    // TODO: fallback hide loader if there iframe.src return 404
+    listenFor(iframe, 'embed-size', data => {
+
+      // We only get the message if and when there is a media to display.
+      this.togglePlaceholder(false);
+      let height = data.interaction.data.size.height;
+      const width = data.interaction.data.size.width;
+      height += (data.interaction.layout.directive === 'contest-poll');
+      iframe.height = height;
+      iframe.width = width;
+      const amp = iframe.parentElement;
+      amp.setAttribute('height', height);
+      amp.setAttribute('width', width);
+      this./*OK*/changeHeight(height);
+    }, /* opt_is3P */false);
+
+
+    // append iframe
+    this.element.appendChild(iframe);
+    return this.iframePromise_ = loadPromise(iframe).then(() => {
+      this.getVsync().mutate(() => {
```

Comment:
Please use vsync with state instead of anon-closures. See [this bug for details](https://github.com/ampproject/amphtml/issues/3718).